### PR TITLE
Fix Spark Delta Lakes links and other improvements in docs

### PR DESCRIPTION
**Pull request by omarsmak**
This change is [Reviewable](https://reviewable.io/reviews/projectnessie/nessie/2028)

### CODE REVIEW

File: site/docs/tools/deltalake/spark.md [hunk 1]
```diff
@@ -3,9 +3,7 @@
 !!! note
     Detailed steps on how to set up Pyspark + Delta Lake + Nessie with Python is available on [Binder](https://mybinder.org/v2/gh/projectnessie/nessie-demos/main?filepath=notebooks/nessie-delta-demo-nba.ipynb).
 
-To access Nessie from a spark cluster make sure the `spark.jars` spark option is set to include
-the [Spark 2](https://repo.maven.apache.org/maven2/org/projectnessie/nessie-deltalake-spark2/{{ versions.java}}/nessie-deltalake-spark2-{{ versions.java}}.jar)
-or [Spark 3](https://repo.maven.apache.org/maven2/org/projectnessie/nessie-deltalake-spark3/{{ versions.java}}/nessie-deltalake-spark3-{{ versions.java}}.jar)
+To access Nessie from a spark cluster make sure the `spark.jars` spark option is set to include the [Spark 3](https://repo.maven.apache.org/maven2/org/projectnessie/nessie-deltalake/{{ versions.java}}/nessie-deltalake-{{ versions.java}}.jar)
```
Comment (dimas-b) [inline site/docs/tools/deltalake/spark.md:6]: just to clarify: Is Spark 2 no longer supported?

Comment (omarsmak) [reply site/docs/tools/deltalake/spark.md:6]: I think so, is it @snazy ? If yes, is it also applicable for Iceberg as I see in the docs we have a section about Spark 2, I wonder if we clean that up as well  🤔

Comment (snazy) [reply site/docs/tools/deltalake/spark.md:6]: Yea, only Deltalake 1.0 w/ Spark 3 is supported. We had clients for DL 0.6 w/ Spark2 + DL 0.8 w/ Spark 3 earlier.
ALl mentions of Deltalake w/ Spark 2 can go away.

Comment (snazy) [reply site/docs/tools/deltalake/spark.md:6]: Note: maybe use `[Nessie Deltalake Client for Spark 3]` instead `[Spark 3]` for the link?

Comment (dimas-b) [reply site/docs/tools/deltalake/spark.md:6]: +1 to link renaming

**Review by nastra — DISMISSED** [review]

File: site/docs/tools/deltalake/spark.md (hunk 1 above)
Comment (omarsmak) [reply site/docs/tools/deltalake/spark.md]: Fix that as well removed other Spark 2 instances in the docs in regards to Delta Lake

File: site/docs/tools/index.md [hunk 2]
```diff
@@ -25,7 +25,7 @@ the following:
 |Create Branch| :material-check: | :material-check: | :material-check:|
 |Create Tag| :material-check: | :material-check: |:material-check:|
 |Iceberg Tables|:material-check:|:material-check:| |
-|Delta Lake Tables|:material-check:|:material-check:| |
+|Delta Lake Tables||:material-check:| |
```
Comment (nastra) [inline site/docs/tools/index.md:28]: should we also remove `Spark2` from the feature matrix here?

Comment (omarsmak) [reply site/docs/tools/index.md:28]: If I am not mistaken, Spark2 is still relevant to Iceberg and thus, just removed the check mark from the Delta Lake / Spark2 matrix.

**Review by dimas-b — APPROVED** [review]

File: site/docs/tools/deltalake/spark.md [hunk 3]
```diff
@@ -98,7 +96,7 @@ Spark support is constantly evolving and the differences in Spark3 vs Spark2.4 i
 
 ### Spark2
```
Comment (nastra) [inline site/docs/tools/deltalake/spark.md:97]: needs to be updated as well?

Comment (omarsmak) [reply site/docs/tools/deltalake/spark.md:97]: I have removed it, I think this title is no longer needed

**Review by nastra — APPROVED** [review]

**Pull request merged**